### ISSUE TITLE
Problem changing between C Euphonium and Euphonium treble clef working with parts

### ISSUE THREAD

**Issue by zorba77** [Jojo-Schmitz/MuseScore#437]
Step to reproduce:
1.  create a new score with Euphonium Bb treble clef
2. Write a D scale 
![image](https://github.com/Jojo-Schmitz/MuseScore/assets/137386945/34ee47c6-afcf-422b-9285-660534762913)
3. Create a part 
4. The scale in the part is correct
![image](https://github.com/Jojo-Schmitz/MuseScore/assets/137386945/cd8908b9-f651-472b-abd0-17ce8dab140c)
5. Change in the score the instrument to eufonio, via right click. Notes in the score are correct
![image](https://github.com/Jojo-Schmitz/MuseScore/assets/137386945/dca32408-7613-4d85-9a84-692951569927)
6. This is the result in the part
![image](https://github.com/Jojo-Schmitz/MuseScore/assets/137386945/d2379009-ade0-4c29-9438-476ad0a6e544)
The playback is correct, the above B is played one octave lower, between A and C in the scale
7. If I change the instrument also in the part this is the result
![image](https://github.com/Jojo-Schmitz/MuseScore/assets/137386945/281fc4ba-4678-445b-b111-829f52bc49d2)
8. and this is the result in the score (the same)
![image](https://github.com/Jojo-Schmitz/MuseScore/assets/137386945/c34ca9d2-cf18-4e28-947a-8b85b3bbed64)
9. If I delete and recreate the part notes are correct, but I loss al the layout

The same happen starting with Euphonium and changing to Euphonium treble clef 
The same happen in the 3.6.2 and older, and with C tuba and Bb tuba treble clef if I remember right

OS: Ubuntu 22.04.4 LTS, Arch.: x86_64, MuseScore version (64-bit): 3.7.0.8678350200, revision: GitHub-Jojo-Schmitz-MuseScore-acbfac0

**Comment by Jojo-Schmitz** [comment]
So not a regression.
Does 4.x behave better/correct? If not, please report at https://github.com/musescore/MuseScore/issues/new/choose

**Comment by zorba77** [comment]
MS 4.2.1 is ok, if I change the instrument in the score, the part has the new instrument with correct notes

**Comment by zorba77** [comment]
Now I tried to undo the instrument change, ad a line break in the part, and change the instrument in the score. The part is ok

**Comment by Jojo-Schmitz** [comment]
Seems to happen for all inmstrumets, changing an instrument in the main score doesn't change it in the part and vice versa.

I'd see this as 'by design' in Mu3 rather than as a bug.

**Comment by zorba77** [comment]
True, but change the instrument in the score corrupt the part, so something is wrong

**Comment by zorba77** [comment]
I'm studying the problem,
It seems that when you change the instrument in the score, notes are changed in the part, but not the instrument type.
In the attached file you can find:
- Eufonio: the first score, with euphonium treble clef in the the main score and in the part
- Eufonio: the score, with Eufonio treble clef changed to Eufonio C in the score (the relative part is wrong) 
- Eufonio DO: the same score, with same notes, but with Eufonio C from scratch

I'm trying to study with kdiff the differences 

[Musescore bug parts.zip](https://github.com/user-attachments/files/16159092/Musescore.bug.parts.zip)

**Comment by zorba77** [comment]
Part of the output of: diff Eufonio1.mscx Eufonio\ DO.mscx
The problem seems that when you change the instrument type in the score, notes in the part are changed (are removed transposing info), but not the information about transposing instrument

261c252
<       <metaTag name="partName">Eufonio (Chiave di violino)</metaTag>
---
>       <metaTag name="partName">Eufonio</metaTag>
268,269c259
<           <defaultConcertClef>F</defaultConcertClef>
<           <defaultTransposingClef>G</defaultTransposingClef>
---
>           <defaultClef>F</defaultClef>
271,272c261,262
<         <trackName>Eufonio (Chiave di violino)</trackName>
<         <Instrument id="euphonium-treble">
---
>         <trackName>Eufonio</trackName>
>         <Instrument id="euphonium">
275c265
<           <trackName>Eufonio (Chiave di violino)</trackName>
---
>           <trackName>Eufonio</trackName>
280,281d269
<           <transposeDiatonic>-8</transposeDiatonic>
<           <transposeChromatic>-14</transposeChromatic>
283,284c271
<           <concertClef>F</concertClef>
<           <transposingClef>G</transposingClef>
---
>           <clef>F</clef>
